OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/AnalyticsProvider.java

```diff
@@ -0,0 +1,327 @@
+package org.edx.mobile.module.analytics;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.inject.Singleton;
+
+import org.edx.mobile.util.images.ShareUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An analytics provider class whose function is to contain all the analytics services we are using
+ * in our app, and call their respective APIs to send out analytics events.
```

Comment:
Since this just contains a list of other implementations of `Analytics`, I suggest rewording the description to something like this:

> A registry for enabled `Analytics` implementations, that delegates all methods to them.

And "provider" can be changed to "registry" or "delegate" as you prefer.